xcodeproj/internal/input_files.bzl

```diff
@@ -32,9 +32,24 @@ def _collect_transitive_extra_files(info):
     inputs = info.inputs
     transitive = [inputs.extra_files]
     if not info.target:
-        transitive.append(inputs.srcs)
-        transitive.append(inputs.non_arc_srcs)
-        transitive.append(inputs.hdrs)
+        new_srcs = []
```

Comment:
This is super verbose, I'm sure there's a nicer way to do this.

Reply:
Hmm, I guess this is why I just had them as `File`.

I wonder if it would be better to have a second `extra_file_paths` instead of changing the type of `extra_files`? Then in `xcodeproj.bzl` we append `extra_file_paths` to the transformed `extra_files` and call `to_dto` on that?

Reply:
I don't feel strongly either way. I have a slight preference for keeping it as-is because I think this mapping is still simpler than having two ways to add extra files.

I was more asking if there's a more concise way to perform this mapping of a depset. Like `inputs.srcs.map(file_path)`.

Reply:
The downside to this way is that we are cracking open the depset for each target, and if there are files that are duplicated in the inputs it's more efficient to only it once later.

I'm fine with it like this though, since I'll be refactoring stuff later. Better to keep it localized for now.

As for your question:
```
transitive.extend([
    file_path(file)
    for file in inputs.srcs.to_list()
])
```

Reply:
Ah I didn't notice this syntax elsewhere, it's surprising to me because it's wrapped in square brackets already. [e53f82f](https://github.com/buildbuddy-io/rules_xcodeproj/pull/301/commits/e53f82ffd9d10de58a5090a9e5b0e339787ee786)